iOS/MobileStarter/react-app/src/ProjectPicker.tsx

```diff
@@ -0,0 +1,143 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import React from "react";
+import { HorizontalPicker, useIsMountedRef } from "@itwin/mobile-ui-react";
+import { Project, ProjectsAccessClient, ProjectsQueryArg, ProjectsQueryFunction, ProjectsSearchableProperty, ProjectsSource } from "@itwin/projects-client";
+import { ProgressCallback } from "@itwin/core-frontend/lib/cjs/request/Request";
+import { IModelApp } from "@itwin/core-frontend";
+import { LoadingSpinner } from "@itwin/core-react";
+import { SearchControl, i18n, presentError, ButtonProps, HubScreenButton, HubScreenButtonListProps, HubScreenButtonList } from "./Exports";
+
+// Get all the projects that this user has access to, sorted by most recent use.
+async function getProjects(progress?: ProgressCallback, source = ProjectsSource.All, searchString = "") {
+  const client = new ProjectsAccessClient();
+  const chunkSize = 100;
+  const allProjects: Project[] = [];
+  const accessToken = await IModelApp.getAccessToken();
+
+  try {
+    let queryArgs: ProjectsQueryArg = { pagination: { skip: 0, top: chunkSize } };
+    if (source === ProjectsSource.All && searchString.length > 0)
+      queryArgs.search = { searchString, propertyName: ProjectsSearchableProperty.Name, exactMatch: false };
+    else
+      queryArgs.source = source;
+    const chunk = await client.getByQuery(accessToken, queryArgs);
+    allProjects.push(...chunk.projects);
+    progress?.({ loaded: allProjects.length });
+    return { projects: allProjects, next: chunk.links?.next };
+  } catch (ex) {
+    console.log(`Exception fetching projects: ${ex}`);
+    if (allProjects.length > 0) {
+      return { projects: allProjects, next: undefined };
+    } else {
+      throw ex;
+    }
+  }
+}
+
+interface ProjectButtonProps extends Omit<ButtonProps, "title"> {
+  project: Project;
+}
+
+function ProjectButton(props: ProjectButtonProps) {
+  const { project, onClick } = props;
+  return <HubScreenButton
+    title={project.name ?? ""}
+    onClick={onClick}
+  />;
+}
+
+interface ProjectListProps extends HubScreenButtonListProps {
+  projects: Project[];
+  onClick?: (project: Project) => void;
```

Comment:
Similar to the iModels list, I don't like using onClick for something that selects an item from a list.

Reply:
Changed to onSelect